### PR TITLE
Edit Genesys and Twilio reference implementation docs

### PR DESCRIPTION
**Pull request by rrberry**

### CODE REVIEW

File: src/middleware/genesys/README.md [hunk 1]
```diff
@@ -1,34 +1,72 @@
...
+
+1. In [`purecloudSetup.ts`](../../serviceDesks/genesys/purecloudSetup.ts), set the value of `AUTH_SERVER_BASE_URL` to the public URL for your server. Make sure you specify the `https://` URL.
+
+1. In the Genesys Cloud user interface, open the settings for your web chat widget and enable **Require Authentication**.
+
+1. In the **Authentication URL** field, specify `https://<server-url>/jwt`, where `<server-url>` is the public URL for your server. Click **Save**.
+
+1. In [`genesysServiceDesk.ts`](../../serviceDesks/genesys/genesysServiceDesk.ts), set the `WIDGET_REQUIRES_AUTHENTICATION` and `AUTHENTICATED_CALLS_ENABLED` flags to `true`.
+
+  **Note:** These flags are independent, so you do not have to enable both simultaneously. For example, if you set `WIDGET_REQUIRES_AUTHENTICATION = false` and `AUTHENTICATED_CALLS_ENABLED = true`, authentication willl be disabled in the web chat widget, but the server will still use an OAuth token to authenticate when accessing agent availability status.
```
Comment (VikramVuppla) [inline src/middleware/genesys/README.md:70]: minor typo in willl

File: src/middleware/genesys/README.md [hunk 2]
```diff
@@ -1,34 +1,72 @@
...
+
+1. From the project root directory, run `npm run dev` again, and test. The server should now authenticate the chat and fetch agent availability.
```
Comment (VikramVuppla) [inline src/middleware/genesys/README.md:72]: numbering incorrect, due to the Note, `From the project root..` is set to 1

File: docs/STEPS.md [hunk 3]
```diff
@@ -2,103 +2,113 @@
...
+    Implementation of this step will vary significantly depending on your service desk provider's API, how it handles authentication, and how you choose to handle authentication for your middleware server.
+
+      For example, a Genesys Cloud integration requires that you set up an OAuth client ](https://help.mypurecloud.com/articles/create-an-oauth-client/) that uses a client credentials grant. You can then make a request to the OAuth client using the Genesys PureCloud [REST API](https://developer.mypurecloud.com/api/rest/authorization/use-client-credentials.html) for an access token. (For more information, see [the Genesys PureCloud tutorial](https://developer.mypurecloud.com/api/tutorials/oauth-client-credentials/?language=nodejs&step=1).) See the Genesys Cloud [reference implementation](../src/middleware/genesys/README.md) for a simple example.
```
[the remark this answers is not in the document]
Comment (rrberry) [reply docs/STEPS.md]: Fixed link

File: docs/STEPS.md [hunk 4]
```diff
@@ -2,103 +2,113 @@
...
 
-    - Implement `areAnyAgentsOnline` in order to check for agent availability.
+    - Implement `areAnyAgentsOnline()` in order to check for agent availability.
 
-    - Provide availability information to the end user using `this.callback.updateAgentAvailability`.
+    - Provide availability information to the end user using `this.callback.updateAgentAvailability()`.
 
     Make sure your error-handling flow includes cases in which it is not possible to connect the end user to an agent (for example, if no agents are available).
 
+      For example, a Genesys Cloud integration would use the PureCloud REST API to [qhttps://developer.mypurecloud.com/api/rest/v2/analytics/queue.html]. (For more information, see the Genesys Cloud [tutoriall](https://developer.mypurecloud.com/api/tutorials/number-of-agent-in-queue/?language=python&step=1).
```
[the remark this answers is not in the document]
Comment (rrberry) [reply docs/STEPS.md]: fixed link and typo ... time to get my MacBook keyboard fixed

File: src/middleware/flex/README.md [hunk 5]
```diff
@@ -1,95 +1,78 @@
...
-`FLEX_FLOW_ID` | Custom Flex flow SID. You need to use the twilio CLI to create a `custom` flex flow as described [here](https://www.twilio.com/blog/add-custom-chat-channel-twilio-flex)
-`API_KEY` | API keys to access Flex APIs. You can generate [here](https://www.twilio.com/console/project/api-keys) in the _Create a new Flex Flow_ section
-`API_SECRET` | secret for the `API_KEY`. 
-`CHAT_SERVICE_ID` | Programmable chat service SID available [here](https://www.twilio.com/console/chat/services)
-##### Install dependencies
+`ACCOUNT_ID` | Your primary Twilio Flex account SID, as shown in the [Twilio console](https://www.twilio.com/console/project/settings).
+`AUTH_TOKEN` | Your Auth Token for the Twilio Flex project, as shown in the [Twilio console](https://www.twilio.com/console/project/settings). (For more information, see [Auth Tokens and How to Change Them](https://support.twilio.com/hc/en-us/articles/223136027-Auth-Tokens-and-How-to-Change-Them).)
+`FLEX_FLOW_ID` | The Custom FlexFlow SID. Use the Twilio CLI to create a custom Flex Flow (for more information, see [Add a Custom Chat Channel to Twilio Flex](https://www.twilio.com/blog/add-custom-chat-channel-twilio-flex).)
```
[the remark this answers is not in the document]
Comment (rrberry) [reply src/middleware/flex/README.md]: Made both parenthetical comments separate sentences, with no parentheses

File: src/middleware/genesys/README.md [hunk 6]
```diff
@@ -1,34 +1,72 @@
...
+
+  You can refer to these Genesys docs and resources for more information about using the Genesys PureCloud API:
+
+    - [Guest Chat Client - JavaScript](https://developer.mypurecloud.com/api/rest/client-libraries/javascript-guest/index.html)
+
+    - [Guest Chat APIs](https://developer.mypurecloud.com/api/webchat/guestchat.html)
+
+    - [Guest Chat Client API documentation](https://developer.mypurecloud.com/api/rest/client-libraries/javascript-guest/WebChatApi.html)
+
+    - [Guest Chat Client API source code](https://github.com/MyPureCloud/purecloud-guest-chat-client-javascript/blob/9599e33609a87358671532b10e53fad24e592373/build/src/purecloud-guest-chat-client/api/WebChatApi.js)
```
[the remark this answers is not in the document]
Comment (rrberry) [reply src/middleware/genesys/README.md]: fixed indentation in Markdown to get correct formatting

File: src/middleware/genesys/README.md (hunk 2 above)
Comment (rrberry) [reply src/middleware/genesys/README.md:69]: added indentation to get correct numbering

**Comment by rrberry** [comment]
All comments should now be addressed.

**Review by VikramVuppla — APPROVED** [review]

**Pull request merged**